### PR TITLE
List/room granularity for the account data extension

### PR DESCRIPTION
**Pull request by DMRobertson**
More of #41.

This was easier than I expected; the hardest part was writing the test.

I'd intended for one of the tests I wrote in #50 to demonstrate that I had correctly changed `AppendLive`, but I think the test I wrote simply doesn't do that.

I think what I've got here really does exercise the `AppendLive` path---but it does so as an e2e test and is probably going to be brittle (e.g. there are timeouts). Maybe this would be better suited for`extensions/account_data_test.go`?

### CODE REVIEW

File: tests-e2e/account_data_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+					},
+				},
+			},
+			WithPos(pos),
+		)
+		pos = response.Pos
+		responses <- response
+	}
+
+	go aliceIncrementalSync()
```
Comment (kegsay) [inline tests-e2e/account_data_test.go:99]: Please don't use goroutines unless you absolutely have to. You don't have to here. It's sufficient to just call `putGlobalAccountData` and then synchronously call `alice.SlidingSyncUntil(...)` to await for the response. This handles timeouts for you.

**Review by kegsay — CHANGES_REQUESTED** [review]
Tests need to be simpler and not have channels/goroutines, otherwise this is excellent, LGTM

File: tests-e2e/account_data_test.go (hunk 1 above)
Comment (DMRobertson) [reply tests-e2e/account_data_test.go:99]: The thinking here: I was trying to test that `AppendLive` has the correct logic in place. If the proxy is already aware of a change before the sync request comes in, that change gets picked up by `ProcessInitial` and a response is returned immediately, without hitting `AppendLive`.

Comment (DMRobertson) [reply tests-e2e/account_data_test.go]: This thinking was not correct. See 391f0b8

**Review by kegsay — APPROVED** [review]

**Pull request merged**